.github/workflows/build.yml

```diff
@@ -1,301 +1,293 @@
+            -   name: Compare bundle stats
+                uses: github/webpack-bundlesize-compare-action@v1
+                if: github.event_name != 'push'
+                continue-on-error: true
+                with:
+                    github-token: ${{ secrets.GITHUB_TOKEN }}
+                    current-stats-json-path: "packages/web3/dist/${{ github.ref_name }}.json"
+                    base-stats-json-path: "packages/web3/dist/4.x.json"
+            -   name: Cache default branch stats
+                uses: actions/cache/save@v3
+                if: github.event_name == 'push' && github.ref == 'refs/heads/4.x'
+                with:
+                    path: packages/web3/dist/4.x.json
+                    key: web3-bundle-stats-4x-${{github.sha}}
 
-  benchmark:
-      name: Benchmark Tests
-      needs: build
-      runs-on: ubuntu-latest
-      strategy:
-          matrix:
-              node: [ 18 ]
-      steps:
-          -   uses: actions/setup-node@v3
+    unit:
+        name: Unit Tests
+        needs: build
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                node: [18, 20]
+        steps:
+            - uses: actions/setup-node@v4
               with:
                   node-version: ${{ matrix.node }}
-          -   uses: actions/download-artifact@v4
+            - uses: actions/cache/restore@v3
               with:
-                  name: web3-${{ matrix.node }}.js.tar.gz
-                  path: /tmp
-          -   run: tar -xf /tmp/web3-${{ matrix.node }}.js.tar.gz -C ./
-              # @octokit/core not supported on node 16, so I can't add it to the package.json
-          -   run: npm install --no-package-lock --no-save --force @octokit/core
-          -   name: Restore main branch benchmark data
-              uses: actions/cache/restore@v3
+                    path: ./
+                    key: web3-${{ matrix.node }}-${{github.event.pull_request.base.sha}}
+            - run: yarn test:unit
+            - name: Upload coverage to Codecov
+              uses: codecov/codecov-action@v3
               with:
-                  path: web3-benchmark-main.json
-                  key: ${{ runner.os }}-web3-benchmark-main.json
-          -   run: yarn test:benchmark
-          -   name: Compare benchmark result and make comment
-              uses: benchmark-action/github-action-benchmark@v1
+                  flags: UnitTests
+                  token: ${{ secrets.CODECOV_TOKEN }}
+              if: ${{ matrix.node == 18 }}
+
+    integration:
+        name: Integration # (ganache with HTTP)
+        needs: build
+        runs-on: ubuntu-latest
+        env:
+            INFURA_GOERLI_HTTP: ${{ secrets.INFURA_GOERLI_HTTP }}
+            INFURA_GOERLI_WS: ${{ secrets.INFURA_GOERLI_WS }}
+        strategy:
+            fail-fast: false
+            matrix:
+                node: [18]
+                backend: ['ganache']
+                mode: ['http']
```

Comment:
there is only one combination of these three so no need of matrix, use these directly in test command